elements/lisk-blocks/src/data_access/storage.ts

```diff
@@ -0,0 +1,265 @@
+		arrayOfBlockIds: ReadonlyArray<string>,
+	): Promise<BlockJSON> {
+		const [block] = await this._storage.entities.Block.get(
+			{
+				id_in: arrayOfBlockIds,
+			},
+			{ sort: 'height:desc', limit: 1 },
+		);
+
+		return block;
+	}
+
+	public async getBlocksCount(): Promise<number> {
+		const count = await this._storage.entities.Block.count({}, {});
+
+		return count;
+	}
+
+	/*
+		Extended blocks with transaction payload
+	*/
+
+	public async getBlocksByIDs(
+		arrayOfBlockIds: ReadonlyArray<string>,
+	): Promise<BlockJSON[]> {
+		const blocks = await this._storage.entities.Block.get(
+			{ id_in: arrayOfBlockIds },
+			{ extended: true },
+		);
+
+		return blocks;
+	}
+
+	public async getBlockByHeight(height: number): Promise<BlockJSON> {
+		const [block] = await this._storage.entities.Block.get(
+			{ height },
+			{ extended: true },
+		);
+
+		return block;
+	}
+
+	public async getBlocksByHeightBetween(
+		fromHeight: number,
+		toHeight: number,
+	): Promise<BlockJSON[]> {
+		const blocks = await this._storage.entities.Block.get(
+			{ height_gte: fromHeight, height_lte: toHeight },
+			// tslint:disable-next-line no-null-keyword
+			{ extended: true, limit: null, sort: 'height:desc' },
+		);
+
+		return blocks;
+	}
+
+	public async getLastBlock(): Promise<BlockJSON> {
+		const [lastBlock] = await this._storage.entities.Block.get(
+			{},
+			{ sort: 'height:desc', limit: 1, extended: true },
+		);
+
+		return lastBlock;
+	}
+
+	public async getTempBlocks(): Promise<TempBlock[]> {
+		const tempBlocks = await this._storage.entities.TempBlock.get(
+			{},
+			{ sort: 'height:asc' },
+		);
+
+		return tempBlocks;
+	}
+
+	public async isTempBlockEmpty(): Promise<boolean> {
+		const isEmpty = await this._storage.entities.TempBlock.isEmpty();
+
+		return isEmpty;
+	}
+
+	public clearTempBlocks(): void {
+		this._storage.entities.TempBlock.truncate();
+	}
+
+	public async getFirstBlockIdWithInterval(
+		height: number,
+		interval: number,
+	): Promise<BlockRound[]> {
+		const rows: BlockRound[] = await this._storage.entities.Block.getFirstBlockIdOfLastRounds(
+			{
+				height,
+				numberOfRounds: 5,
+				numberOfDelegates: interval,
+			},
+		);
+
+		return rows;
+	}
+
+	public async isBlockPersisted(blockId: string): Promise<boolean> {
+		const isPersisted = await this._storage.entities.Block.isPersisted({
+			blockId,
+		});
+
+		return isPersisted;
+	}
+
+	/*
+		Accounts
+	*/
+
+	public async getAccountsByPublicKey(
+		arrayOfPublicKeys: ReadonlyArray<string>,
+		tx?: StorageTransaction,
+	): Promise<Account[]> {
+		const accounts = await this._storage.entities.Account.get(
+			{ publicKey_in: arrayOfPublicKeys },
+			{},
+			tx,
+		);
+
+		return accounts;
+	}
+
+	public async getAccountsByAddress(
+		arrayOfAddresses: ReadonlyArray<string>,
+		tx?: StorageTransaction,
+	): Promise<Account[]> {
+		const accounts = await this._storage.entities.Account.get(
+			{ address_in: arrayOfAddresses },
+			{},
```

Comment:
in general, we should always put `limit` like `limit: arraoOfAddress.length` or `limit: null` when we are getting multiple results.